src/Microsoft.ML.LightGBM/LightGbmArguments.cs

```diff
@@ -129,54 +132,114 @@ internal static class Defaults
                 public bool UnbalancedSets = false;
 
+                /// <summary>
+                /// The minimum loss reduction required to make a further partition on a leaf node of the tree.
+                /// </summary>
+                /// <value>
+                /// Larger values make the algorithm more conservative.
+                /// </value>
                 [Argument(ArgumentType.AtMostOnce,
                     HelpText = "Minimum loss reduction required to make a further partition on a leaf node of the tree. the larger, " +
                         "the more conservative the algorithm will be.")]
                 [TlcModule.Range(Min = 0.0)]
                 public double MinimumSplitGain = 0;
 
+                /// <summary>
+                /// The maximum depth of a tree.
+                /// </summary>
+                /// <value>
+                /// 0 means no limit.
+                /// </value>
                 [Argument(ArgumentType.AtMostOnce,
                     HelpText = "Maximum depth of a tree. 0 means no limit. However, tree still grows by best-first.")]
                 [TlcModule.Range(Min = 0, Max = int.MaxValue)]
                 public int MaximumTreeDepth = 0;
 
+                /// <summary>
+                /// The minimum sum of instance weight needed to form a new node.
+                /// </summary>
+                /// <value>
+                /// If the tree partition step results in a leaf node with the sum of instance weight less than <see cref="MinimumChildWeight"/>,
+                /// the building process will give up further partitioning. In linear regression mode, this simply corresponds to minimum number
+                /// of instances needed to be in each node. The larger, the more conservative the algorithm will be.
+                /// </value>
                 [Argument(ArgumentType.AtMostOnce,
                     HelpText = "Minimum sum of instance weight(hessian) needed in a child. If the tree partition step results in a leaf " +
                         "node with the sum of instance weight less than min_child_weight, then the building process will give up further partitioning. In linear regression mode, " +
                         "this simply corresponds to minimum number of instances needed to be in each node. The larger, the more conservative the algorithm will be.")]
                 [TlcModule.Range(Min = 0.0)]
                 public double MinimumChildWeight = 0.1;
 
+                /// <summary>
+                /// The frequency of performing subsamplig (bagging).
+                /// </summary>
+                /// <value>
+                /// 0 means disable bagging. N means perform bagging at every N iterations.
+                /// To enable bagging, <see cref="SubsampleFraction"/> should also be set to a value less than 1.0.
```

Comment:
>should also be set to a value less than 1.0 [](start = 71, length = 43)

Just curious, where did this information come from?

Reply:
three sources: 
*[https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst](https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst)
*[https://lightgbm.readthedocs.io/en/latest/Parameters.html](https://lightgbm.readthedocs.io/en/latest/Parameters.html)
*ML.NET source code

---
In reply to: [263689038](https://github.com/dotnet/machinelearning/pull/2886#discussion_r263689038) [](ancestors = 263689038)